sbt-plugin/src/main/scala/org/scalajs/sbtplugin/ScalaJSPluginInternal.scala

```diff
@@ -908,21 +908,35 @@ object ScalaJSPluginInternal {
       scalaJSModuleInitializers :=
         (scalaJSModuleInitializers in (This, Zero, This)).value,
 
-      scalaJSModuleInitializers ++= {
+      scalaJSModuleInitializers ++= Def.taskDyn {
         if (scalaJSUseMainModuleInitializer.value) {
-          Seq(scalaJSMainModuleInitializer.value.getOrElse {
-            throw new MessageOnlyException(
-                "No main module initializer was specified (possibly because " +
-                "no or multiple main classes were found), but " +
-                "scalaJSUseMainModuleInitializer was set to true. " +
-                "You can explicitly specify it either with " +
-                "`mainClass := Some(...)` or with " +
-                "`scalaJSMainModuleInitializer := Some(...)`")
-          })
+          Def.task {
+            Seq(scalaJSMainModuleInitializer.value.getOrElse {
+              scalaJSDiscoveredMainClasses.value.keys.toSeq match {
```

Comment:
Consider using
```scala
if (scalaJSDiscoveredMainClasses.value.isEmpty)
```
instead of `.keys.toSeq match { ... }`

Reply:
Fixed in https://github.com/scala-js/scala-js/pull/3724/commits/3a78772b5bc3ad9576b894710c5759fe755f9cf2